src/api/services/iuf/sessions.go

```diff
@@ -508,13 +522,29 @@ func (s iufService) getDagTasks(session iuf.Session, stageInfo iuf.Stage) ([]v1a
 	if stageInfo.Type == "product" {
 		for _, product := range session.Products {
 			for index, operation := range stageInfo.Operations {
+				if !templateMap[operation.Name] {
+					s.logger.Warnf("The template %v cannot be found in Argo. Make sure you have run upload-rebuild-templates.sh from docs-csm", operation.Name)
+					continue
+				}
+
 				task := v1alpha1.DAGTask{
 					Name: product.Name + "-" + operation.Name,
 				}
 				// dep with a stage
 				if index != 0 {
-					task.Dependencies = []string{
-						product.Name + "-" + stageInfo.Operations[index-1].Name,
+					// check the last dependency is available
+					var lastDependencyOpName string
```

Comment:
Alternatively to this code, would it just work to set the `lastDependencyOpName` at the end of this loop and then look at it on the next operation? That way we wouldn't have to keep iterating over all the previous operations for each operation.

Reply:
Good point

Reply:
Done and unit tested.